File: api/agents.py
from fastapi import APIRouter

from agents.orchestrator import AGENT_TASKS

router = APIRouter()

AGENT_LABELS = {
    "legal_score": "Score",
    "recommendations": "Recs",
    "summary": "Summary",
    "tone_analysis": "Tone",
    "risk_analysis": "Risk",
    "clause_analysis": "Clause",
}


@router.get("/agents/status")
def get_agents_status():
    return {
        "backend": "online",
        "agents": {
            name: "ready"
            for name in AGENT_TASKS
        },
    }

File: frontend/src/lib/agents.js
export const AGENT_LIST = [
  { id: "legal_score", label: "Score" },
  { id: "recommendations", label: "Recs" },
  { id: "summary", label: "Summary" },
  { id: "tone_analysis", label: "Tone" },
  { id: "risk_analysis", label: "Risk" },
  { id: "clause_analysis", label: "Clause" },
];

export const INITIAL_AGENT_STATUS = Object.fromEntries(
  AGENT_LIST.map((a) => [a.id, "idle"])
);

export const STATUS_LABELS = {
  idle: "Idle",
  offline: "Offline",
  ready: "Ready",
  running: "Running",
  done: "Done",
  error: "Error",
};
